client/src/main/java/org/eclipse/hono/client/AsyncCommandClient.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client;
+
+import java.util.Map;
+
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * A client for sending asynchronous request response commands, i.e. all command responses for a specific tenant and
+ * reply ID shall be received by the same consumer in an asynchronous fashion.
+ * <p>
+ * An instance of this interface is always scoped to a specific tenant and device.
+ * </p>
+ *
+ * @see org.eclipse.hono.client.HonoClient#createAsyncCommandResponseConsumer(String, String,
+ *      java.util.function.Consumer, io.vertx.core.Handler)
+ * @see org.eclipse.hono.client.HonoClient#createAsyncCommandResponseConsumer(String, String,
+ *      java.util.function.BiConsumer, io.vertx.core.Handler)
+ */
+public interface AsyncCommandClient extends MessageSender {
+
+    /**
+     * Sends an async command to a device, i.e. there is no immediate response expected from the device, but
+     * asynchronously via a separate consumer.
+     * <p>
+     * A device needs to be (successfully) registered before a client can upload any data for it. The device also needs
+     * to be connected for a successful delivery.
+     *
+     * @param command The command name.
+     * @param data The command data to send to the device or {@code null} if the command has no input data.
+     * @param correlationId The identifier to use for correlating the response with the request. Note: This parameter
+     * is security sensitive. To ensure secure request response mapping choose correlationId carefully, e.g.
+     * {@link java.util.UUID#randomUUID()}.
+     * @param replyId An arbitrary string which gets used for response link address in the form of <em>control/${
+     * tenantId}/${replyId}</em> Must match the {@code replyId} passed to the command response receiver, see also below.
```

Comment:
below? where?
Didn't you want to add a warning that using *device ID* as the *replyId* is strongly discouraged and might lead to unexpected results?

Reply:
I was referring to the @see links below. If this is confusing I can just remove the "see below" hint.

About the warning I am not sure. From my point of view this is a bug that will be fixed with #1073. If you like I can add a hint, but there will be the danger that we miss to remove it after the fix of #1073.

Reply:
that is true, then let's not add the hint.